packages/node/src/trpc.ts

```diff
@@ -0,0 +1,44 @@
+import { getCurrentHub } from '@sentry/core';
+import { normalize } from '@sentry/utils';
+
+interface SentryTrpcMiddlewareOptions {
+  attachRpcInput?: boolean;
+}
+
+interface TrpcMiddlewareArguments<T> {
+  path: string;
+  type: 'query' | 'mutation' | 'subscription';
+  next: () => Promise<T>;
+  rawInput: unknown;
+}
+
+/**
+ * Sentry tRPC middleware that names the handling transaction after the called procedure.
+ *
+ * Use the Sentry tRPC middleware in combination with the Sentry server integration. (e.g. express integration or
+ * Next.js SDK)
+ */
+export async function sentryTrpcMiddleware(options: SentryTrpcMiddlewareOptions) {
+  return async function <T>({ path, type, next, rawInput }: TrpcMiddlewareArguments<T>): Promise<T> {
+    const hub = getCurrentHub();
+    const clientOptions = hub.getClient()?.getOptions();
+    const sentryTransaction = hub.getScope()?.getTransaction();
+
+    if (sentryTransaction) {
+      sentryTransaction.setName(`${path}()`, 'route');
+      sentryTransaction.op = 'rpc.server';
```

Comment:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md I don't see `rpc.server` operation in the spec we follow, there's `rpc` only. Do we make a distinction here ourselves?

Reply:
In general, I think it is fine to *append* stuff to the span op. I think there is a hierarchy to it. The docs are just a guide to set the baseline ops we should use.

Here I decided to append `.server` because we could at some point decide that we would like to instrument the client side part of RPCs and there we would probably call it `rpc.client`. Similar to how we do it with `http.server` and `http.client`.

Reply:
Yeah this is fine, we can update our span operations spec to account for this https://develop.sentry.dev/sdk/performance/span-operations/